redis/asyncio/connection.py

```diff
@@ -291,19 +288,21 @@ async def _read_response(
         elif byte == b":":
             response = int(response)
         # bulk response
-        elif byte == b"$":
-            length = int(response)
-            if length == -1:
-                return None
-            response = await self._read(length)
+        elif byte == b"$" and response == b"-1":
+            return None
+        elif byte == b"$" and response != b"-1":
+            response = await self._read(int(response))
         # multi-bulk response
-        elif byte == b"*":
-            length = int(response)
-            if length == -1:
-                return None
+        elif byte == b"*" and response == b"-1":
+            return None
+        elif byte == b"*" and response != b"-1":
             response = [
-                (await self._read_response(disable_decoding)) for _ in range(length)
+                (await self._read_response(disable_decoding))
+                for _ in range(int(response))  # noqa
             ]
+        elif byte not in (b"-", b"+", b":", b"$", b"*"):
```

Comment:
Maybe this line could be just `elif byte != b"+":`. All other comparisons are known to be true here because none of the previous ifs/elifs matched.